lib/web/mage/validation.js

```diff
@@ -396,6 +396,24 @@
             $.mage.__('Please enter at least {0} characters')
         ],
 
+        /* detect chars that would require more than 3 bytes */
+        'charbytes': [
+            function (value) {
+                var validator = this,
+                    message = $.mage.__('Please remove invalid characters: {0}.'),
+                    matches = value.match(/(?:[\uD800-\uDBFF][\uDC00-\uDFFF])/g),
```

Comment:
How is that different from https://stackoverflow.com/a/16496799? Could you provide a source for current regexp maybe?

Reply:
https://stackoverflow.com/a/16346705
https://en.wikipedia.org/wiki/UTF-16#Code_points_U.2B10000_to_U.2B10FFFF

Future reference solution won't work as it's only for ES6 and we're currently ES5.

Reply:
Thanks, looks good to me then :)